crates/astria-sequencer-types/src/namespace.rs

```diff
@@ -41,15 +41,21 @@ impl Namespace {
     /// returns an 10-byte namespace given a byte slice by hashing
     /// the bytes with sha256 and returning the first 10 bytes.
     #[must_use]
+    #[allow(clippy::missing_panic_docs)]
     pub fn from_slice(bytes: &[u8]) -> Namespace {
         let mut hasher = Sha256::new();
         hasher.update(bytes);
         let result = hasher.finalize();
+        const _: () = assert!(
+            NAMESPACE_ID_AVAILABLE_LEN <= 32,
```

Comment:
I believe celestia limits us to much less than 32 bytes? https://celestiaorg.github.io/celestia-app/specs/namespace.html

28 bytes of which we can only use 10?

I'm a bit confused of the purpose here, we have an assert on a constant?

Reply:
This is compile time stuff and just to ensure that the `expect` right below is correct. I can't get the `32` bytes out of `sha256` so I need to hardcode 32 here.

Just best practices because `NAMESPACE_ID_AVAILABLE_LEN` is imported into this crate and could just be changed without us noticing.